File: app/src/main/java/com/scatl/uestcbbs/module/houqin/adapter/HouQinReportTopicImageAdapter.java
package com.scatl.uestcbbs.module.houqin.adapter;

import android.util.Log;
import android.view.View;
import android.widget.ImageView;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;

/**
 * author: sca_tl
 * date: 2020/10/25 17:58
 * description:
 */
public class HouQinReportTopicImageAdapter extends BaseQuickAdapter<String, BaseViewHolder> {
    public HouQinReportTopicImageAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, String item) {
        helper.addOnClickListener(R.id.image);
        GlideLoader4Common.simpleLoad(mContext, item+"", helper.getView(R.id.image));
        helper.getView(R.id.delete_btn).setVisibility(View.GONE);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/houqin/adapter/HouQinReportTopicImageAdapter.kt
package com.scatl.uestcbbs.module.houqin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemCreateCommentImageBinding
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 15:31
 */
class HouQinReportTopicImageAdapter: PreloadAdapter<String, ItemCreateCommentImageBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemCreateCommentImageBinding {
        return ItemCreateCommentImageBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemCreateCommentImageBinding>, position: Int, item: String?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }
        holder.binding.image.load(item)
        holder.binding.deleteBtn.visibility = View.GONE
    }
}